lib/pages/portal/service/website_provider.dart

```diff
@@ -395,4 +397,21 @@ class WebsiteProvider with ChangeNotifier {
       return false;
     }
   }
+
+  Future<bool> uploadWebPicture(Website website, Uint8List file) async {
+    final result = await StorageProvider.uploadImage(
+        file, 'websites/${website.id}/icon.png');
+    if (!result) {
+      if (file.length > 5 * 1024 * 1024) {
+        AppToast.show(S.current.errorPictureSizeToBig);
+      } else {
+        AppToast.show(S.current.errorSomethingWentWrong);
+      }
+    }
+    return result;
+  }
+
+  Future<String> getWebPictureURL(Website website) {
+    return StorageProvider.findImageUrl('websites/${website.id}/icon.png');
+  }
```

Comment:
Optional: This could potentially be an extension. Something like this, added to the top of this file:

```
extension IconURLExtension on Website {
  Future<String> getIconURL() =>  StorageProvider.findImageUrl('websites/${id}/icon.png');
}
```

This can allow you to call `website.getIconURL()` directly.

Either way works, I just wanted to point out an alternative. Your call.